homework-submission/App.js

```diff
@@ -0,0 +1,105 @@
+const path = '/data.json';
+
+const ListItem = ({ id, description, deadline, className }) => {
+  return (
+    <li className={className}>
+      Todo List Number: {id} Description: {description} Deadline: {deadline}
+    </li>
+  );
+};
+
+class TodoApp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: [],
+    };
+  }
+
+  componentDidMount() {
+    readFile(path).then(response => {
+      this.setState(() => ({
+        todos: response,
+      }));
+    });
+  }
+
+  render() {
+    const todos = this.state.todos;
+    const NewTodo = {
+      id: 0,
+      description: '',
+      deadline: '',
+      done: false,
+    };
+    let TodoNumber = 0;
+
+    return (
+      <div>
+        <h2>TodoLists</h2>
+        <ul>
+          {todos.map(({ id, description, deadline, done }, i) => (
+            <ListItem
+              id={id}
+              description={description}
+              deadline={deadline}
+              className={done === true ? 'strike_through' : ''}
+              key={i}
+            />
+          ))}
+        </ul>
+        <form>
```

Comment:
This form can get a Component on it's own

Reply:
Divided into components.